pkg/cmd/server/server.go

```diff
@@ -867,23 +892,20 @@ func (s *server) runControllers(defaultVolumeSnapshotLocations map[string]string
 		s.logger.WithField("informer", informer).Info("Informer cache synced")
 	}
 
-	// now that the informer caches have all synced, we can start running the controllers
 	for i := range controllers {
 		controllerRunInfo := controllers[i]
-
-		wg.Add(1)
-		go func() {
-			controllerRunInfo.controller.Run(ctx, controllerRunInfo.numWorkers)
-			wg.Done()
-		}()
+		// Adding the controllers to the manager will register them as a (runtime-controller) runnable,
+		// so the manager will ensure the cache is started and ready before all controller are started
+		s.mgr.Add(managercontroller.Runnable(controllerRunInfo.controller, controllerRunInfo.numWorkers))
 	}
 
-	s.logger.Info("Server started successfully")
+	s.logger.Info("Server starting...")
 
-	<-ctx.Done()
+	if err := s.mgr.Start(s.ctx.Done()); err != nil {
```

Comment:
`err` should be returned

Reply:
I'm doing a `s.logger.Fatal` so I don't think it's needed?

Reply:
Yeah, I would think if the error's handled, it shouldn't also be returned.